ui/app/src/model/config-client.ts

```diff
@@ -190,6 +190,7 @@ export interface DatasourceConfig {
 }
 
 export interface ConfigModel {
+  api_prefix?: string;
```

Comment:
Sorry I overlooked the fact that this uses a fetch to get the config. so if the API has a prefix fetching the config won't work as at that moment the prefix is not known. One option is to set this value globally on the window object. And serve a Javascript from the backend with the value. From react we can read that value, same as the plugins will do for the assets path.

Reply:
Oh, I missed this too. Ok, yeah, let me take a look at something.

I'm thinking that if the go backend pulls the values from the config and injects them into the HTML then they'll be up to date on reload. The alternative is replacing them at runtime, but that means the container needs a restart if anything changes.

What do you think?

Reply:
I'd say we can add the minimal params so the config can be fetched, in this case we just need the apiPrefix. If the config changes the frontend will just do a new request. Maybe is better to serve a Javascript file rather than parse the whole HTML. WDYT?

Reply:
Ended up using rspack since it's already being used to do the same thing for the public path assets:
[Just these changes are here](https://github.com/perses/perses/pull/3377/files/c699eee3a076fdc54e0214ade17c16fc8adacd34..6e06ab3d3a9fd5f566a1395ec627d79cd2ee0fca)